### ISSUE TITLE
Inefficient iteration

### ISSUE THREAD

**Issue by hariedo** [adammyhre/Unity-Improved-Timers#1]
I noticed that your UpdateTimers() was allocating a virgin empty List<Timer> on every tick, every Update.  I know you're doing this to allow Timer.Tick() to register new timers, or deregister itself, which would modify the list. 
However, if you have more than a handful of timers, this can cause a fair amount of allocation churn, as each new List<T>() has to start small, then grow a number of times to finally match the original list just for the purpose of iteration.

Adding a second sweep list will keep the internal capacity stable and avoid the churn.

```
        private static readonly List<Timer> timers = new();
        private static readonly List<Timer> sweep = new();

        public static void UpdateTimers() {
            sweep.Clear();
            sweep.AddRange(timers);
            foreach (var timer in sweep)
                timer.Tick();
        }
```

In addition, if we consider the "improved" timer manager as being the *owner* of timer instances, this owner has every right and a responsibility to be the one to dispose their resources when the list is cleared.

```
        public static void Clear()
        {
            int count = timers.Count;
            sweep.Clear();
            sweep.AddRange(timers);
            foreach (var timer in sweep)
                timer.Dispose();
            timers.Clear();
            sweep.Clear();
            Debug.Log($"TimerManager cleared {count} timers");
        }
```

This will (1) actually dispose of resources at the time Unity stops play mode when you Clear() the list, instead of the next domain reload, and (2) make the .Dispose() call completely optional on the part of callers which create new timers.  Expecting every caller to remember to .Dispose() themselves is iffy at best, especially since the object model is so unlike a regular C#/CLI/process.

Thoughts?

**Comment by adammyhre** [comment]
These seem like good improvements.  I'll give it some more thought today and see about implementing your suggestions tonight.

**Comment by adammyhre** [comment]
Could add a guard clause before performing any operations in the `UpdateTimers` method to avoid any unnecessary operations.
```csharp
if (timers.Count == 0) return;
```

And the 2 lines that Clear and AddRange could be moved into a method or an extension method.  Not sure on the name yet.

```csharp
public static class ListExtensions {
    public static void UpdateWith<T>(this List<T> list, IEnumerable<T> items) {
        list.Clear();
        list.AddRange(items);
    }
}
```

**Comment by adammyhre** [comment]
Resolves #1
